File: src/main/java/com/gorkem/vehicle_inspector/entity/Vehicle.java
package com.gorkem.vehicle_inspector.entity;

import jakarta.persistence.*;

@Entity
@Table(name = "vehicles")
public class Vehicle {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 20)
    private String plate;

    @Column(nullable = false, length = 50)
    private String brand;

    @Column(nullable = false, length = 50)
    private String model;

    @Column(name = "model_year", nullable = false)
    private Integer modelYear;

    @Column(nullable = false)
    private Integer mileage;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(
            name = "user_id",
            nullable = false,
            foreignKey = @ForeignKey(
                    name = "fk_vehicle_user"
            )
    )
    private User user;

    protected Vehicle() {
    }

    public Vehicle(
            String plate,
            String brand,
            String model,
            Integer modelYear,
            Integer mileage,
            User user
    ) {
        this.plate = plate;
        this.brand = brand;
        this.model = model;
        this.modelYear = modelYear;
        this.mileage = mileage;
        this.user = user;
    }

    public Long getId() {
        return id;
    }

    public String getPlate() {
        return plate;
    }

    public String getBrand() {
        return brand;
    }

    public String getModel() {
        return model;
    }

    public Integer getModelYear() {
        return modelYear;
    }

    public Integer getMileage() {
        return mileage;
    }

    public User getUser() {
        return user;
    }

    public void setPlate(String plate) {
        this.plate = plate;
    }

    public void setBrand(String brand) {
        this.brand = brand;
    }

    public void setModel(String model) {
        this.model = model;
    }

    public void setModelYear(Integer modelYear) {
        this.modelYear = modelYear;
    }

    public void setMileage(Integer mileage) {
        this.mileage = mileage;
    }

    public void setUser(User user) {
        this.user = user;
    }
}

File: mobile/lib/features/vehicle/models/vehicle.dart
class Vehicle {
  const Vehicle({
    required this.id,
    required this.plate,
    required this.brand,
    required this.model,
    required this.modelYear,
    required this.mileage,
  });

  final int id;
  final String plate;
  final String brand;
  final String model;
  final int modelYear;
  final int mileage;

  String get displayName => '$brand $model';

  factory Vehicle.fromJson(
      Map<String, dynamic> json,
      ) {
    return Vehicle(
      id: (json['id'] as num?)?.toInt() ?? 0,
      plate: json['plate'] as String? ?? '',
      brand: json['brand'] as String? ?? '',
      model: json['model'] as String? ?? '',
      modelYear:
      (json['modelYear'] as num?)?.toInt() ?? 0,
      mileage:
      (json['mileage'] as num?)?.toInt() ?? 0,
    );
  }
}